src/core/etl/src/Flow/ETL/DataFrame.php

```diff
@@ -793,11 +795,22 @@ public function select(string|Reference ...$entries) : self
      */
     public function sortBy(Reference ...$entries) : self
     {
-        $this
-            ->cache($this->context->config->id())
-            ->run();
+        try {
+            $extractor = (new MemorySort(new PipelineExtractor($this->pipeline), $this->context->config->sortMemoryLimit()))->sortBy($this->context, refs(...$entries));
+        } catch (OutOfMemoryException $e) {
+            $extractor = (new ExternalSort(
+                $this->context->config->id(),
+                $this->context->cache(),
+                new PipelineExtractor($this->pipeline),
+                new ExternalSort\RowCache\FilesystemSortRowCache(
+                    $this->context->filesystem(protocol('file')),
+                    $this->context->config->serializer()
+                ),
+                100
```

Comment:
this is the output batch size of the after sort, we need to think from where to take this value

Reply:
or add new argument `$batchSize` to `\Flow\ETL\DataFrame::sortBy` method?

Reply:
I resolved this one by detecting the current batch size while moving rows to sort cache